delivery_tnt_oca/models/tnt_request.py

```diff
@@ -291,6 +291,9 @@ def _prepare_create_shipping(self):
                 },
             },
         }
+        # This may be optional if the customer has fixed pick-up.
+        if self.activity_book:
+            data["ACTIVITY"]["BOOK"] = {"CONREF": self.record.name}
```

Comment:
What is `self.record.name`? There's no options on this?

Reply:
`self.record.name` = `picking.name`

According to tnt support, in some specific cases it is not necessary to send the BOOK key.

Reply:
But explain what is `BOOK` key. `tnt_activity_book` field name is not saying anything about its purpose.

Reply:
`Finally, BOOK` tag is not necessary.